packages/grafana-runtime/src/measurement/collector.ts

```diff
@@ -1,212 +1,84 @@
-  measurements = new Map<string, MeasurementCache>();
-  config: MeasurementCacheConfig = {
-    append: 'tail',
-    capacity: 600, // Default capacity 10min @ 1hz
+  measurements = new Map<string, StreamingDataFrame>();
+  config: StreamingFrameOptions = {
+    maxLength: 600, // Default capacity 10min @ 1hz
   };
 
   //------------------------------------------------------
   // Public
   //------------------------------------------------------
 
   getData(query?: MeasurementsQuery): DataFrame[] {
-    const { name, labels, fields } = query || {};
-
-    let data: DataFrame[] = [];
-    if (name) {
-      // for now we only match exact names
-      const m = this.measurements.get(name);
-      if (m) {
-        data = m.getFrames(labels);
+    const { key, fields } = query || {};
+
+    // Find the data
+    let data: StreamingDataFrame[] = [];
+    if (key) {
+      const f = this.measurements.get(key);
+      if (!f) {
+        return [];
       }
+      data.push(f);
     } else {
+      // Add all frames
       for (const f of this.measurements.values()) {
-        data.push.apply(data, f.getFrames(labels));
+        data.push(f);
       }
     }
 
+    // Filter the fields we want
     if (fields && fields.length) {
       let filtered: DataFrame[] = [];
       for (const frame of data) {
         const match = frame.fields.filter((f) => fields.includes(f.name));
         if (match.length > 0) {
-          filtered.push({ ...frame, fields: match }); // Copy the frame with fewer fields
+          filtered.push({ ...frame, fields: match, length: frame.length }); // Copy the frame with fewer fields
         }
       }
+      if (filtered.length) {
+        return filtered;
+      }
     }
     return data;
   }
 
-  getDistinctNames(): string[] {
+  getKeys(): string[] {
     return Object.keys(this.measurements);
   }
 
-  getDistinctLabels(name: string): Labels[] {
-    const m = this.measurements.get(name);
-    if (m) {
-      return Object.keys(m.frames).map((k) => parseLabels(k));
-    }
-    return [];
-  }
-
-  setCapacity(size: number) {
-    this.config.capacity = size;
-
-    // Now update all the circular buffers
-    for (const wrap of this.measurements.values()) {
-      for (const frame of Object.values(wrap.frames)) {
-        for (const field of frame.fields) {
-          (field.values as CircularVector).setCapacity(size);
-        }
-      }
-    }
-  }
-
-  getCapacity() {
-    return this.config.capacity!;
-  }
-
-  clear() {
-    this.measurements.clear();
+  ensureCapacity(size: number) {
+    // TODO...
   }
 
   //------------------------------------------------------
   // Collector
   //------------------------------------------------------
 
-  addBatch = (batch: MeasurementBatch) => {
-    let action = batch.action ?? MeasurementAction.Append;
-    if (action === MeasurementAction.Clear) {
-      this.measurements.clear();
-      action = MeasurementAction.Append;
+  addBatch = (msg: MeasurementBatch) => {
```

Comment:
@FZambia this is a temporary hack while we sync the frontend and backend implementations.  Currently on the backend, we have types for `DataFrameJSON`, but not a batch of them -- we are also missing `key` on DataFrameJSON and that will be required if we want to support multiple frames in one channel